### PR TITLE
chore: adjust directory structure of api

### LINKED ISSUE #721: refactory(manager-api): use Golang standard file structure
we need to follow Golang standard way(unofficial, but more reasonable):

https://github.com/golang-standards/project-layout

the current file structure is unacceptable:

image

### LINKED ISSUE #731: [style] adjust directory structure of api to golang standard style
ref to #715

### PR DESCRIPTION
**Pull request by gxthrj**
Fix #721 
Fix #731 

- [x] Added `cmd` path and move the `main.go` to `./api/cmd/manager`.
- [x] Moved `filter` to `internal/filter`
- [x] Moved `log` to `internal/log`
- [x] Moved `conf/conf.go` to `internal/conf/conf.go`

Ref to [golang-standards-project-layout](https://github.com/golang-standards/project-layout)

### CODE REVIEW

**Comment by nic-chen** [comment]
CI failed. @gxthrj

**Comment by gxthrj** [comment]
> CI failed. @gxthrj

fixed

**Comment by nic-chen** [comment]
should we move 'log' and 'conf' into 'internal' too？

**Comment by gxthrj** [comment]
> should we move 'log' and 'conf' into 'internal' too？

Done

File: api/internal/filter/authentication.go [hunk 1]
```diff
@@ -20,11 +20,11 @@ import (
...
 	"strings"
 
-	"github.com/dgrijalva/jwt-go"
 	"github.com/gin-gonic/gin"
 
-	"github.com/apisix/manager-api/conf"
-	"github.com/apisix/manager-api/log"
+	"github.com/apisix/manager-api/internal/conf"
+	"github.com/apisix/manager-api/internal/log"
+	jwt "github.com/dgrijalva/jwt-go"
```
Comment (starsz) [inline api/internal/filter/authentication.go:27]: I think we should classify the import package.

See https://github.com/uber-go/guide/blob/master/style.md#import-group-ordering

File: api/internal/filter/recover.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 	"runtime"
 	"time"
 
-	"github.com/apisix/manager-api/log"
+	"github.com/apisix/manager-api/internal/log"
```
Comment (starsz) [inline api/internal/filter/recover.go:27]: Ditto.

File: api/internal/handler/route/route.go [hunk 3]
```diff
@@ -30,15 +30,15 @@ import (
...
-	"github.com/apisix/manager-api/conf"
+	"github.com/apisix/manager-api/internal/conf"
 	"github.com/apisix/manager-api/internal/core/entity"
 	"github.com/apisix/manager-api/internal/core/store"
 	"github.com/apisix/manager-api/internal/handler"
+	"github.com/apisix/manager-api/internal/log"
 	"github.com/apisix/manager-api/internal/utils"
 	"github.com/apisix/manager-api/internal/utils/consts"
-	"github.com/apisix/manager-api/log"
+	lua "github.com/yuin/gopher-lua"
```
Comment (starsz) [inline api/internal/handler/route/route.go:41]: Ditto.

**Comment by nic-chen** [comment]
conflicts ..

File: api/internal/handler/route/route.go (hunk 3 above)
Comment (gxthrj) [reply api/internal/handler/route/route.go]: Done

File: api/internal/filter/recover.go (hunk 2 above)
Comment (gxthrj) [reply api/internal/filter/recover.go]: Done

File: api/internal/filter/authentication.go (hunk 1 above)
Comment (gxthrj) [reply api/internal/filter/authentication.go]: Done

File: api/internal/handler/route/route.go [hunk 4]
```diff
@@ -29,16 +29,16 @@ import (
...
+	wgin "github.com/shiningrush/droplet/wrapper/gin"
+	"github.com/apisix/manager-api/internal/conf"
 	"github.com/apisix/manager-api/internal/core/entity"
 	"github.com/apisix/manager-api/internal/core/store"
 	"github.com/apisix/manager-api/internal/handler"
+	"github.com/apisix/manager-api/internal/log"
 	"github.com/apisix/manager-api/internal/utils"
 	"github.com/apisix/manager-api/internal/utils/consts"
-	"github.com/apisix/manager-api/log"
+
```
Comment (starsz) [inline api/internal/handler/route/route.go:41]: Redundant line.

Comment (gxthrj) [reply api/internal/handler/route/route.go]: Done

**Comment by nic-chen** [comment]
@gxthrj  do we need to update docs for changes ?

**Comment by gxthrj** [comment]
> @gxthrj do we need to update docs for changes ?

The original document refers to `make build` directly when building, so there is no need to update docs.
But we can add a supplementary notes on directory structure.

**Review by tokers — APPROVED** [review]

**Comment by gxthrj** [comment]
cc @nic-chen

**Review by nic-chen — APPROVED** [review]

**Pull request merged**